src/Middleware.php

```diff
@@ -3,31 +3,138 @@
+        // we'll attempt to use the Laravel Mix asset manifest to generate our hash.
+        if (file_exists($manifest = public_path('mix-manifest.json'))) {
+            return md5_file($manifest);
         }
+    }
+
+    /**
+     * Defines the props that are shared by default.
+     *
+     * @see https://inertiajs.com/shared-data
+     * @param  \Illuminate\Http\Request  $request
+     * @return array
+     */
+    public function share(Request $request)
+    {
+        return [
+            'errors' => function () use ($request) {
+                return $this->resolveValidationErrors($request);
+            },
+        ];
+    }
+
+    /**
+     * Handle the incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  Closure  $next
+     * @return Response
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        Inertia::share($this->share($request));
+
+        Inertia::version(function () use ($request) {
+            return $this->version($request);
+        });
+
+        $response = $next($request);
+        $response = $this->checkVersion($request, $response);
+        $response = $this->changeRedirectCode($request, $response);
+
+        return $response;
+    }
 
-        if ($request->method() === 'GET' && $request->header('X-Inertia-Version', '') !== Inertia::getVersion()) {
+    /**
+     * In the event that the asset version changes, Inertia will automatically reload
+     * the page in order to ensure the application keeps working for the end user.
+     *
+     * @param  Request  $request
+     * @param  Response  $response
+     * @return Response
+     */
+    public function checkVersion(Request $request, Response $response)
+    {
+        if ($request->header('X-Inertia') &&
+            $request->method() === 'GET' &&
+            $request->header('X-Inertia-Version', '') !== Inertia::getVersion()
+        ) {
             if ($request->hasSession()) {
                 $request->session()->reflash();
             }
 
-            return Response::make('', 409, ['X-Inertia-Location' => $request->fullUrl()]);
+            return new Response('', 409, ['X-Inertia-Location' => $request->fullUrl()]);
         }
 
-        if ($response instanceof Redirect && $response->getStatusCode() === 302 && in_array($request->method(), ['PUT', 'PATCH', 'DELETE'])) {
+        return $response;
+    }
+
+    /**
+     * Changes the status code during an Inertia redirect, so that the browser
+     * makes the request as GET instead, avoiding unsupported method calls.
+     *
+     * @param  Request  $request
+     * @param  Response  $response
+     * @return Response
+     */
+    public function changeRedirectCode(Request $request, Response $response)
+    {
+        if ($response instanceof RedirectResponse &&
+            $request->header('X-Inertia') &&
+            $response->getStatusCode() === 302 &&
```

Comment:
Why are we event checking for the `RedirectResponse` class here? All we really care about is if the status code is `302`, which we already check.

```suggestion
        if ($request->header('X-Inertia') &&
            $response->getStatusCode() === 302 &&
```

Reply:
That's.. a very good point.